app/assets/javascripts/components/overview/milestones.jsx

```diff
@@ -28,9 +30,11 @@ const Milestones = createReactClass({
     const weekNumberOffset = CourseDateUtils.weeksBeforeTimeline(this.props.course);
     const blocks = [];
 
-    this.props.allWeeks.map((week) => {
+    this.props.allWeeks.map((week,navIndex) => {
```

Comment:
Where does this come from? In Timeline.jsx, it the result of some complex code in the render cycle for handling empty weeks.

Reply:
The `navIndex` is generated by the `map` function, which helps identify unique elements in the array. Additionally, the `DateCalculator` requires the `navIndex` prop for its calculations. Upon cross-checking in the Timeline page, I confirmed that the `navIndex` is simply the index number generated by the `map` function.